File: backend/src/main/java/com/ameri/objects/beans/adminBeans/GananciaBeans.java
package com.ameri.objects.beans.adminBeans;

import java.util.List;

public class GananciaBeans {

    private double total;
    private List<OtherMagazineBeans> otherMagazineBeans;

    {

        total = 0;
    }

    public GananciaBeans( List<OtherMagazineBeans> otherMagazineBeans) {
        this.otherMagazineBeans = otherMagazineBeans;
    }

    public void updateTotal(){
        for(OtherMagazineBeans other: otherMagazineBeans){
            total += other.getGananciaTotal().doubleValue();
            System.out.println(other.getGananciaTotal().doubleValue());
        }
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public List<OtherMagazineBeans> getOtherMagazineBeans() {
        return otherMagazineBeans;
    }

    public void setOtherMagazineBeans(List<OtherMagazineBeans> otherMagazineBeans) {
        this.otherMagazineBeans = otherMagazineBeans;
    }

    public double getTotal() {
        return total;
    }
}

File: frontend/src/objects/classes/beans/adminBeans/GananciaBeans.ts
import {OtherMagazineBeans} from "./OtherMagazineBeans";

export class GananciaBeans{
  total: number;
  otherMagazineBeans: Array<OtherMagazineBeans>

  constructor(total: number, otherMagazineBeans: Array<OtherMagazineBeans>) {
    this.total = total;
    this.otherMagazineBeans = otherMagazineBeans;
  }
}
